Add GEOS-IT option for remap restarts

biljanaorescanin opened the pull request:
I added remapping from GEOS-IT restarts. 
GEOS-IT has only two dates available for users to be remapped, day 14th and day 28th, and they are in .tar format.

This PR will address these two issues:
https://github.com/GEOS-ESM/GEOS_Util/issues/62 
https://github.com/GEOS-ESM/GEOS_Util/issues/72

PR was tested for a run using the on-screen option for GEOS-IT vs. copying and un-tar files manually and using the tool to remap my restarts. The difference is zero diff.

mathomp4 commented:
Well that was a lot more than I was expecting for this! Great job!

weiyuan-jiang commented:
> PR was tested for diff input source restarts; M2, GEOS-IT and user owned restarts remapping. PR was also tested with automated tests on SLES12 and that is zero diff as well.
> 
> @weiyuan-jiang I've added agcm rst remap since I didn't want to confuse users if we didn't do that with GEOS-IT, this way it is same principal as M2 handling.

The problem is if the users answer No to remap agcm_import _rst, it will be copied and remapped anyway.

biljanaorescanin commented:
> > PR was tested for diff input source restarts; M2, GEOS-IT and user owned restarts remapping. PR was also tested with automated tests on SLES12 and that is zero diff as well.
> > @weiyuan-jiang I've added agcm rst remap since I didn't want to confuse users if we didn't do that with GEOS-IT, this way it is same principal as M2 handling.
> 
> The problem is if the users answer No to remap agcm_import _rst, it will be copied and remapped anyway.

Good catch, I've fixed it.

biljanaorescanin commented:
I've found trying out different dates loop was not good. I was testing just with resent dates. Now it works from 2007 01.

biljanaorescanin commented:
@weiyuan-jiang testing case `c24Toc12` will fail since we now set false for default for `agcm_import_rst` remapping.  It was just that base case did have that remapped and current doesn't so file count comparison fails. 

But we could just remove that restarts from base? Or change default ? 

All other testing was ok.  I didn't find any more issues.

gmao-rreichle commented:
cc @gmao-rreichle
